psyneulink/core/components/mechanisms/mechanism.py

```diff
@@ -3027,123 +3027,167 @@ def _gen_llvm_ports(self, ctx, builder, ports, group,
-                                       mech_params, mech_state, mech_input)
+        builder = self._gen_llvm_ports(ctx,
+                                       builder,
+                                       param_ports,
+                                       "_parameter_ports",
+                                       _get_modulated_param_output_ptr,
+                                       _get_param_base_ptr,
+                                       mech_params,
+                                       mech_state,
+                                       mech_input)
         return params_out, builder
 
-    def _gen_llvm_output_port_parse_variable(self, ctx, builder,
-                                             mech_params, mech_state, value, port):
+
+    def _gen_llvm_output_port_parse_variable(self, ctx, builder, mech_params, mech_state, value, port):
+        """Create output port variable based on the variable specification."""
         port_spec = port._variable_spec
-        if port_spec == OWNER_VALUE:
-            return value
-        elif port_spec == OWNER_EXECUTION_COUNT:
-            # Convert execution count to (num_executions, TimeScale.LIFE)
-            # The difference in Python PNL is that the former counts across
-            # all contexts. This is not possible in compiled code, thus
-            # the two are identical.
+
+        # Convert "execution_count" to (num_executions, TimeScale.LIFE)
+        # The difference in Python PNL is that the former counts across
+        # all contexts. This is not possible in compiled code, thus
+        # the two are identical.
+        if port_spec == OWNER_EXECUTION_COUNT:
             port_spec = ("num_executions", TimeScale.LIFE)
 
-        try:
-            name = port_spec[0]
-            ids = (x() if callable(x) else getattr(x, 'value', x) for x in port_spec[1:])
-        except TypeError as e:
-            # TypeError means we can't index.
-            # Convert this to assertion failure below
-            data = None
-        else:
-            #TODO: support more spec options
-            if name == OWNER_VALUE:
-                data = value
-            elif name in self.llvm_state_ids:
-                data = ctx.get_param_or_state_ptr(builder, self, name, state_struct_ptr=mech_state)
+        canonical_port_spec = _canonicalize_port_variable_specification(port_spec)
+        assert canonical_port_spec is not None, "Unsupported variable spec: {}".format(port_spec)
+
+        parsed = []
+        for spec in canonical_port_spec:
+            param_name, indices = spec
+
+            if param_name == VALUE:
+                base = value
+            elif param_name in self.llvm_state_ids:
+                base = ctx.get_param_or_state_ptr(builder, self, param_name, state_struct_ptr=mech_state)
             else:
-                data = None
+                assert False, "Unsupported variable spec Parameter: {}".format(param.name)
```

Comment:
```suggestion
                assert False, "Unsupported variable spec Parameter: {}".format(param_name)
```

probably? F821 popped here

Reply:
yeah, thanks for catching it.